### PR TITLE
Masq 196 nonce

### PR DESCRIPTION
**Pull request by leventdem**

### CODE REVIEW

File: src/library/masq.js [hunk 1]
```diff
@@ -531,6 +549,10 @@ class Masq {
     if (!this.masterKey) throw Error('MasterKey is not set')
   }
 
+  _checkNonce () {
+    if (!this.nonce) throw Error('nonce is not set')
+  }
```
Comment (sujeebant) [inline src/library/masq.js]: Check if _checkNonce and _checkMK are necessary as masq-common already throws the error

File: src/library/masq.js [hunk 2]
```diff
@@ -74,8 +74,9 @@ class Masq {
 
     const protectedMK = await this._getProtectedMK()
     try {
-      const MK = await decryptMasterKey(passphrase, protectedMK)
-      this.masterKey = await importKey(MK)
+      const { masterKey, nonce } = await decryptMasterKey(passphrase, protectedMK)
+      this.masterKey = await importKey(masterKey)
```
Comment (G-Ray) [inline src/library/masq.js:78]: is it cleared once `closeProfile()` ?

File: src/library/masq.js [hunk 3]
```diff
@@ -138,7 +141,7 @@ class Masq {
     await db.putAsync('/profile/protectedMK', protectedMK)
     // We do not use this.encryptAndPut method because the master key
     // is not stored in this.profileDB (only set in openProfile)
-    await put(db, this.masterKey, '/profile', privateProfile)
+    await put(db, this.masterKey, '/profile', privateProfile, nonce)
```
Comment (G-Ray) [inline src/library/masq.js]: I think @sujeebant already made a comment about this, but what about passing the nonce as third parameter instead ?

Comment (leventdem) [reply src/library/masq.js]: yes, I had not yet updated masq-app. I do it right now.

Comment (leventdem) [reply src/library/masq.js]: done

File: src/library/masq.js (hunk 2 above)
Comment (leventdem) [reply src/library/masq.js:78]: no, added nonce clear. +1

File: package.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
     "classnames": "^2.2.6",
     "compressorjs": "^1.0.5",
     "detect-browser": "^3.0.1",
-    "masq-common": "git+https://github.com/QwantResearch/masq-common",
+    "masq-common": "git+https://github.com/QwantResearch/masq-common#MASQ-191-hashKeys",
```
Comment (G-Ray) [inline package.json]: ok, please merge masq-common before, then use the master branch here before merging this PR

File: src/containers/Apps/Apps.jsx [hunk 5]
```diff
@@ -36,7 +36,7 @@ class Apps extends PureComponent {
         <p className='subtitle'>Retrouvez vos applications synchronisées avec Masq</p>
         {apps.map((app, index) => (
           <div key={index} className={styles.Card}>
-            <Card minHeight={64} title={app.name} image={app.image} color='#a3005c' description={app.description} />
+            <Card minHeight={64} title={app.name} image={app.imageURL} color='#a3005c' description={app.description} />
```
Comment (G-Ray) [inline src/containers/Apps/Apps.jsx]: great

File: src/library/masq.js (hunk 1 above)
Comment (leventdem) [reply src/library/masq.js]: yes removed.

File: package.json (hunk 4 above)
Comment (leventdem) [reply package.json]: ok

**Review by sujeebant — APPROVED** [review]

**Review by G-Ray — APPROVED** [review]

**Pull request merged**